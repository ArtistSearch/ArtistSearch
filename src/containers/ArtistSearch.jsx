
import React, { useEffect, useState } from 'react';
import SearchControls from '../components/artistSearch/SearchControls';
import SearchList from '../components/artistSearch/SearchList';
import Loading from '../components/Loading';
import { getArtists, getArtistsPage } from '../services/getArtists';

export default function ArtistsSearch() {
  const [loading, setLoading] = useState(false);
  const [pageOffset, setOffset] = useState(0);
  const [pages, setPages] = useState(0);
  const [currentPage, setCurrentpage] = useState(1);
  const [newSearch, setNewSearch] = useState('');
  const [artists, setArtists] = useState([]);
  const [search, setSearch] = useState('');

  useEffect(() => {
    setLoading(true);
    getArtistsPage(search)
      .then(setPages);
    getArtists(search, pageOffset)
      .then(setArtists)
      .finally(() => setLoading(false));
  }, [search, pageOffset]);

  const handleSearch = (e) => {
    setNewSearch(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    setSearch(newSearch);
    setCurrentpage(1);
  };

  const onPageDown = () => {
    setOffset(pageOffset - 20);
    setCurrentpage(currentPage - 1);
  };

  const onPageUp = () => {
    setOffset(pageOffset + 20);
    setCurrentpage(currentPage + 1);
  };

  // eslint-disable-next-line keyword-spacing
  if (loading) return (
    <Loading />
  );

  return (
    <>
      <SearchControls
        search={newSearch}
        onChange={handleSearch}
        onSubmit={handleSubmit}
        onDecrementPage={onPageDown}
        onIncrementPage={onPageUp}
        currentPage={currentPage}
        totalPages={pages}
      />
      <SearchList artists={artists} />
    </>
  );
}
