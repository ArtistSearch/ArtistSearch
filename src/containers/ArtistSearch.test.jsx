/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ArtistSearch from './artistSearch';

describe('Artist search Container', () => {
  it('displays a list of artist from user input', async () => {
    render(<ArtistSearch />);

    const ul = await screen.findByRole('list', { name: 'artists' });
    expect(ul).toBeEmptyDOMElement();

    /*const input = await screen.findByLabelText('Search Artists');
        fireEvent.input(input, 'Beatles');

        // eslint-disable-next-line max-len
        const submitButton = screen.findByRole('button', { name: 'search-artists' });
        fireEvent.submit(submitButton);

        */
  });
});
