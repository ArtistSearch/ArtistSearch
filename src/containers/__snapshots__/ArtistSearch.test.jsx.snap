// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist search Container displays a list of artist from user input 1`] = `
<ul
  aria-label="artists"
/>
`;
